analyses/telomerase-activity-prediction/05-Comparing-MolecularSubtypes-EXTENDScores.R

```diff
@@ -0,0 +1,51 @@
+suppressPackageStartupMessages({
+  library(tidyverse)
+  library(dplyr)
+})
+
+# Comparing EXTEND Scores with histology molecular subtypes 
+# base directories
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git")) 
+analysis_dir <- file.path(root_dir, "analyses", "telomerase-activity-prediction")
+
+# source function
+source(file.path(analysis_dir, "util", "boxplot_by_molecular_subtype.R"))
+
+# histology file
+hist_file <- file.path(root_dir, "data", "pbta-histologies.tsv") 
+
+# telomerase activity-prediction for Stranded_FPKM data
+telomerase_scores <- file.path(analysis_dir, "results", "TelomeraseScores_PTBAStranded_FPKM.txt") 
+
+# output file
+output_file <- file.path(analysis_dir, "plots", "PBTA_MolecularSubtypes.pdf")
+
+## Reading the clinical data
+hist_file <- readr::read_tsv(hist_file, guess_max = 10000)    
+hist_file <- hist_file %>%
+  dplyr::rename("SampleID" = "Kids_First_Biospecimen_ID")
+
+## Reading Stranded FPKM telomerase scores
+telomerase_scores <- read.delim(telomerase_scores)  
+
+### Merging Clinical data with the Telomerase scores
+telomerase_scores <- hist_file %>%
+  dplyr::select(SampleID, short_histology, broad_histology, molecular_subtype) %>%
+  inner_join(telomerase_scores, by = "SampleID")
+
+# filter NA, to be classified and molecular subtype count < 2 (required for p-value calculations)
+telomerase_scores <- telomerase_scores %>%
+  filter(!is.na(molecular_subtype),
+         !grepl("To be classified", molecular_subtype)) %>%
+  group_by(molecular_subtype) %>%
+  mutate(n = n()) %>%
+  filter(n > 1)
+
+# apply function
+# create boxplot of NormEXTENDScores per molecular subtype per histology 
+pdf(output_file, width = 7, height = 5)
```

Comment:
Based on the Google Slides deck included in this comment https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1142#issuecomment-901058603, it does seem like some of these will get included in a multipanel figure. (For context, the multipanel figures in this project are all assembled in [`figures`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/0f5a7263af2b02f80c6949f9d33730fe006eaab2/figures).)

To make that a little easier downstream, I'd recommend making a couple tweaks here:

* We can leverage the fact that `ggsave()` is smart enough to detect the file extension. So instead of using `pdf()` or `png()` we would control the output with file names and use `ggsave()` in `boxplot_by_molecular_subtype()`.
* Save individual files per histology, where the filename reflects the histology.

Reply:
@jaclyn-taroni Sounds good to me, just a qq: I can change to individual plots + ggsave but did you prefer png or pdf?

Reply:
I think PNG for now makes sense.

Reply:
Actually just saw all pngs being used so I used png extension for now. I will change the theme once informed. Thank you!